src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -96,7 +97,9 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
     Entity lectureEntity = LectureUtil.createEntity(lectureName, videoUrl, videoId.get());
     datastore.put(lectureEntity);
     try {
-      initializeTranscript(lectureEntity);
+      String transcriptLanguage =
+          Optional.ofNullable(request.getParameter(PARAM_LANGUAGE)).orElse("");
+      initializeTranscript(lectureEntity, transcriptLanguage);
```

Comment:
Is `PARAM_LANGUAGE` required / checked to be non-null in `validatePostRequest`?  That would simplify this

Will the API fail when passed `""`?  If so, I think we should just return an error in validate when it is null

Reply:
Yes, it will still fail because no language is selected. I was considering doing that, but we check for errors and return in the beginning of the method. If there's no language, we should still be able to create a lecture.